Automattic/wp-calypso
# Purchases: Reduxify Stored Cards on Edit Card Details page

This pull request is a follow-up of https://github.com/Automattic/wp-calypso/pull/6613, https://github.com/Automattic/wp-calypso/pull/6628, https://github.com/Automattic/wp-calypso/pull/6665, https://github.com/Automattic/wp-calypso/pull/6710 that updates the `Edit Card Details` page to use the global state tree to manage credit cards (stored during previous purchases or when a user added a new payment method to an existing purchase) instead of the former Flux store:

![screenshot](https://cloud.githubusercontent.com/assets/594356/16811920/ed13bdb8-492a-11e6-9211-c6e78bd12300.png)

It also fixes the prefilling of the card holder name that was broken.

#### Testing instructions
 
1. Run `git checkout update/reduxify-edit-card-details` and start your server, or open a [live branch](https://calypso.live/?branch=update/reduxify-edit-card-details)
2. Sandbox the store
3. Purchase e.g. a [plan](http://calypso.localhost:3000/plans) or a [site redirect](http://calypso.localhost:3000/domains/add/site-redirect) with a fake credit card
4. Open the [`Purchases` page](http://calypso.localhost:3000/purchases)
5. Click on the new purchase
6. Click on the `Edit Payment Method` action
7. Check that the `Name on Card` field contains the card holder name used in the checkout
8. Enter a new fake credit card (with a different name) and click the `Save Card` button
9. Check that you are redirected back to the `Manage Purchase` page with a success message
10. Check that you can see this new card on the `Billing History` page
11. Navigate back to the same `Edit Card Details` page
12. Check that the `Name on Card` field is prefilled in with the new name

You may want to check other payment methods, or type of purchases. You could also force a browser refresh on certain pages to make sure it doesn't make a difference.

#### Additional notes

I opted not to update the Redux selector to remap keys similarly to what the former assembler was doing because as you can see it here the only value we care is the card holder name. We use the keys provided by the API in any other places where we display credit cards. We can address that in a separate pull request though if most of us think it's worth it.

#### Reviews
 
 
@Automattic/sdev-feed
